MovieQuiz/Presentation/StatisticServiceImplementation.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  StatisticServiceImplementation.swift
+//  MovieQuiz
+//
+//  Created by Toto Tsipun on 14.11.2022.
+//
+
+import Foundation
+
+final class StatisticServiceImplementation: StatisticService {
+    
+    private enum Keys: String {
+        case correct, total, bestGame, gamesCount
+    }
+    
+    var bestGame: GameRecord {
+        get {
+            guard let data = userDefaults.data(forKey: Keys.bestGame.rawValue),
+                  let record = try? JSONDecoder().decode(GameRecord.self, from: data) else {
+                return .init(correct: 0, total: 0, date: Date())
+            }
+            return record
+        }
+        set {
+            guard let data = try? JSONEncoder().encode(newValue) else {
+                print("Unable to save value")
+                return
+            }
+            userDefaults.set(data, forKey: Keys.bestGame.rawValue)
+        }
+    }
+    var totalAccuracy: Double {
+        get {
+            let getCorrect = Double((userDefaults.integer(forKey: Keys.correct.rawValue)))
+            let getTotal = Double((userDefaults.integer(forKey: Keys.total.rawValue)))
```

Comment:
1) Лишние скобки:
2) можно же тоже вынести в свойства класса, и например заприватить. А обращаться уже как correct и total
```suggestion
            let getCorrect = Double(userDefaults.integer(forKey: Keys.correct.rawValue))
            let getTotal = Double(userDefaults.integer(forKey: Keys.total.rawValue))
```

Reply:
Александра, можете помочь, при вынесении в свойства класса выдает ошибку инициализации, что-то не пойму, необходимо их инициализировать?
"Cannot use instance member 'userDefaults' within property initializer; property initializers run before 'self' is available"

Reply:
<img width="1109" alt="Снимок экрана 2022-11-25 в 14 31 57" src="https://user-images.githubusercontent.com/89274127/203976923-e6310117-ae69-4d86-9471-d841b6229c69.png">

Reply:
Нужно добавить свойства с гетерами сеттерами, по аналогии gamesCount, gameRecord)

```
private(set) var correct: Double {
 get { } 
 set { } 
}
```

Reply:
И использовать сразу userDefaults.double (или float я не помню)

Reply:
private(set) – это модификатор, который запрещает изменять свойство вне класса, только читать. Можно просто private

Reply:
Спасибо за подсказку, если свойства вычисляемые, то им не нужен инициализатор))